spring-xd-dirt/src/main/java/org/springframework/xd/dirt/server/admin/deployment/zk/DeploymentSupervisor.java

```diff
@@ -459,6 +456,9 @@ public void takeLeadership(CuratorFramework client) throws Exception {
 				jobDeployments = instantiatePathChildrenCache(client, Paths.JOB_DEPLOYMENTS);
 				jobDeployments.start(PathChildrenCache.StartMode.BUILD_INITIAL_CACHE);
 
+				deploymentQueueForConsumer = new DeploymentQueue(client, deploymentMessageConsumer, Paths.DEPLOYMENT_QUEUE,
+						executorService);
+				deploymentQueueForConsumer.start();
```

Comment:
The exception was thrown because `ZKDeploymentHandler.moduleDeploymentRequests` was null. This reference is set when `onSupervisorElected` is invoked. That is invoked on line 468.

However, the deployment queue is still started before this event is fired. This shortens the window for this condition but it does not eliminate it.

I think I would move the queue creation and start to the bottom of this method (just before the sleep).